### PR TITLE
Theora fixes: proper library versioning, fix build on x86 with assembly enabled

### PR DESCRIPTION
**Pull request by tp-m**
Various fixes for the Meson build discovered as part of GStreamer's binary packaging in https://gitlab.freedesktop.org/gstreamer/cerbero/-/merge_requests/1065

### CODE REVIEW

**Comment by tp-m** [comment]
cc @amyspark

File: subprojects/packagefiles/theora/lib/meson.build [hunk 1]
```diff
@@ -170,6 +170,8 @@ libtheoradec = library(
   vs_module_defs: '../win32/xmingw32/libtheoradec-all.def',
   c_args: compile_args,
   link_args: theoradec_version_flag,
+  soversion: theoradec_soversion,
+  version: theoradec_libversion,
```
Comment (eli-schwartz) [inline subprojects/packagefiles/theora/lib/meson.build:174]: I don't think we need both the soversion and the version, do we? The soversion defaults to the first dotted component of the version, which is what most projects do (including this one).

File: subprojects/packagefiles/theora/lib/meson.build [hunk 2]
```diff
@@ -224,6 +225,8 @@ libtheora = library(
   c_args: compile_args,
   link_args: theora_version_flag,
   vs_module_defs: 'theora.def', # Missing in the official tarballs
+  soversion: theora_soversion,
```
Comment (nirbheek) [inline subprojects/packagefiles/theora/lib/meson.build:228]: `soversion:` is redundant here (and above). If `version:` is specified, the first component is set as `soversion:`.

Comment (nirbheek) [reply subprojects/packagefiles/theora/lib/meson.build:228]: Also need `darwin_versions:` set to the following:
```
libtheora.0.dylib (compatibility version 4.0.0, current version 4.10.0)
libtheoradec.1.dylib (compatibility version 3.0.0, current version 3.4.0)
libtheoraenc.1.dylib (compatibility version 3.0.0, current version 3.2.0)
```

(copying from the gstreamer issue so it's updated in both places)

File: subprojects/packagefiles/theora/lib/meson.build (hunk 1 above)
Comment (tp-m) [reply subprojects/packagefiles/theora/lib/meson.build]: Fixed, thanks.

File: subprojects/packagefiles/theora/lib/meson.build (hunk 2 above)
Comment (tp-m) [reply subprojects/packagefiles/theora/lib/meson.build]: Updated accordingly (I think).

**Review by nirbheek — APPROVED** [review]

**Comment by neheb** [comment]
No releases.json bump

**Pull request merged**